File: MY_REPOS/Lambda-Resource-Static-Assets/2-resources/BLOG/Projects/project-archive/binary-tree-solution/binary-tree-solution/tree_node.py
class TreeNode:
    def __init__(self, value):
        self.value = value
        self.left = None
        self.right = None


node_a = TreeNode("a")
# print(node_a.value)         # 'a'
# print(node_a.right)         # None
# print(node_a.left)          # None

node_b = TreeNode("b")
node_a.left = node_b
# print(node_a.left.value)    # 'b'
# print(node_a.right)         # None

File: _MY_ORGS/Web-Dev-Collaborative/blog-research/ciriculumn/week.8-/week8/binary-tree/lib/tree_node.js
// ! create a treenode constructor with given properties
class TreeNode {
  constructor(val) {
    this.val = val;
    this.left = null;
    this.right = null;
  }
}

root = new TreeNode("a");
let b = new TreeNode("b");
let c = new TreeNode("c");
let d = new TreeNode("d");
let e = new TreeNode("e");
let f = new TreeNode("f");
root.left = b;
root.right = c;
b.left = d;
b.right = e;
c.right = f;

module.exports = {
  TreeNode,
};
